src/impls.rs

```diff
@@ -326,7 +326,10 @@ impl Deserial for Address {
 
 impl<'a> Serial for ContractName<'a> {
     fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
-        self.get_chain_name().serial(out)
+        let name = self.get_chain_name();
+        let len: u16 = name.len().try_into().map_err(|_| Default::default())?;
```

Comment:
I would just remove the `try_into` and use `as` since any value constructed should be shorter. ANd it is undefined what happens if you use `unchecked`.

Reply:
Ok